xtokens/src/lib.rs

```diff
@@ -394,6 +394,53 @@ pub mod module {
 
 			Self::do_transfer_multiassets(who, assets.clone(), fee.clone(), dest, dest_weight, true)
 		}
+
+		/// Transfer currency and use relay-chain asset as fee
+		#[pallet::weight(Pallet::<T>::weight_of_transfer(currency_id.clone(), *amount, dest))]
+		#[transactional]
+		pub fn transfer_using_relaychain_as_fee(
+			origin: OriginFor<T>,
+			currency_id: T::CurrencyId,
+			amount: T::Balance,
+			dest: Box<VersionedMultiLocation>,
+			dest_weight: Weight,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let dest: MultiLocation = (*dest).try_into().map_err(|()| Error::<T>::BadVersion)?;
+
+			let location: MultiLocation = T::CurrencyIdConvert::convert(currency_id.clone())
+				.ok_or(Error::<T>::NotCrossChainTransferableCurrency)?;
+			let asset: MultiAsset = (location, amount.into()).into();
+
+			let fee = MultiAsset {
+				id: AssetId::Concrete(MultiLocation::new(1, Junctions::Here)),
```

Comment:
`orml-xtokens` shouldn't assume the location of a particular asset. Make it a type in config.

Reply:
changes in `do_transfer_multiassets `